### DISCUSSION TITLE
Using the context of a request for each new conn

### DISCUSSION THREAD

**Discussion by Jorge-Guillermo** [go-pg/pg#1944] [Q&A] [answered]
I'm trying to use go-pg as persistence layer in a service which uses a REST API.
When creating a new connection to the database, we though it would make sense to use the context of the REST request involved, so we could easily:
- Get the tracing info from the request context
- End the query if the request was canceled

However we are seeing some ugly race conditions, but before reporting them as issues I was wondering if this approach is not considered a good practice and we should be using a different context for the connections.

**Comment by vmihailenco** [comment, marked as answer]
That is a standard practice if I understand you correctly. You should not see any race conditions.

**Reply by Jorge-Guillermo** [reply]
Thank you so much for the answer, this set me on the right path. 
Because our solution is multitenant and we are using postgres RLS, we are running a "logout" query as safety measure to prevent a new request to use the permissions of a old one by accident (since connections are reused). This query was being launched after returning the request, so naturally the context was already closed, hence the race condition. Interestingly, the queries failing were not the "logouts" themselves but the first queries of the next request which reused the connection.
